src/Common/tests/System/Net/VirtualNetwork/VirtualNetworkStream.cs

```diff
@@ -92,16 +105,15 @@ public override void Write(byte[] buffer, int offset, int count)
             _network.WriteFrame(_isServer, innerBuffer);
         }
 
-        private void UpdateReadStream()
+        public override Task<int> ReadAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken)
         {
-            if (_readStream != null && (_readStream.Position < _readStream.Length))
-            {
-                return;
-            }
+            return Task.FromResult<int>(this.Read(buffer, offset, count));
```

Comment:
Not at this point but I'd like to evolve the virtual network to simulate those as well. Adding this and `cancellationToken` support.